packages/eslint-plugin/docs/rules/strict-enums.md

```diff
@@ -0,0 +1,122 @@
+# `strict-enums`
+
+Disallows the usage of unsafe enum code patterns.
+
+## Rule Details
+
+The TypeScript compiler will allow you to set any number to a variable containing a number enum, like this:
+
+```ts
+enum Fruit {
+  Apple,
+  Banana,
+}
+
+let fruit = Fruit.Apple;
+fruit = 999; // No error
+```
+
+This rule flags the following unsafe enum practices:
```

Comment:
@bradzacher @Zamiell it just struck me that there's a parallel between these _unsafe_ practices and our `no-unsafe-*` rules around `any`! What do you think about splitting this rule up into more granular rules? Starting name suggestions:

* `no-unsafe-enum-value-changes`: enum incrementing/decrementing / `incorrectIncrement`
* `no-unsafe-enum-assignments`: mismatched enum declarations/assignments / `mismatchedAssignment`
* `no-unsafe-enum-comparisons`: mismatched enum comparisons / `mismatchedComparison`
* `no-unsafe-enum-parameters`: mismatched enum function arguments / `mismatchedFunctionArgument`

Two thoughts in support of this:
* We had a hard time naming this rule because it's so general. That on its own is an indication it might want to be split up.
* Each of these behaviors might want to be configured granularly. It'd be easier to do that when they're in individual rules.

I'm happy to do the splitup work as part of getting this PR in.

Reply:
From https://github.com/typescript-eslint/typescript-eslint/pull/6091#issuecomment-1327548775:

> Hey Josh,
> 
> > What do you think about splitting this rule up into more granular rules?
> 
> I'm fine with splitting `no-unsafe-enum-value-changes` into a separate rule (that errors on any increment/modification operator). However, I think the others should be in one rule because they are essentially doing the same check (e.g. `number !== enum`).

I've played around this a bit more and think I understand better. It feels like there are two kinds of operations in play:

* `no-unsafe-enum-assignments`: e.g. `fruit++`
* `no-unsafe-enum-comparisons`: e.g. `fruit === vegetable`

🤔 now I'm less sure about splitting into two rules...

Reply:
I slept on it and am now more certain on it being two rules 😄.